nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AbstractExecuteSQL.java

```diff
@@ -273,10 +274,21 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                     throw failure.getRight();
                 }
 
+                final Map<String, String> sqlParameters = context.getProperties()
+                        .entrySet()
+                        .stream()
+                        .filter(e -> e.getKey().isDynamic())
+                        .collect(Collectors.toMap(e -> e.getKey().getName(), Map.Entry::getValue));
+
                 if (fileToProcess != null) {
-                    JdbcCommon.setParameters(st, fileToProcess.getAttributes());
+                    sqlParameters.putAll(fileToProcess.getAttributes());
+                }
+
+                if (!sqlParameters.isEmpty()) {
+                    JdbcCommon.setParameters(st, sqlParameters);
```

Comment:
The `JdbcCommon.setParameters()` method logs parameter values and includes the parameter value in some exception messages, which is a problem for sensitive properties.